pkg/netstore/netstore.go

```diff
@@ -36,7 +46,10 @@ var (
 
 // New returns a new NetStore that wraps a given Storer.
 func New(s storage.Storer, validStamp postage.ValidStampFn, rcb recovery.Callback, r retrieval.Interface, logger logging.Logger) storage.Storer {
-	return &store{Storer: s, validStamp: validStamp, recoveryCallback: rcb, retrieval: r, logger: logger}
+	ns := &store{Storer: s, validStamp: validStamp, recoveryCallback: rcb, retrieval: r, logger: logger}
+	ns.sCtx, ns.sCancel = context.WithCancel(context.Background())
```

Comment:
instead of having context in the struct, you could have a quit channel instead. continued....

Reply:
This is definitely one way of doing it. But there is a context required for the operation in the routines created. This is why I prefer using a background context. Achieves the same purpose and is pretty clear from the code.